File: PDOO/NapakalakiRuby/bad_consequence.rb
a# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
#encoding: utf-8
  
class BadConsequence
    attr_accessor :text, :levels, :nVisibleTreasures, :nHiddenTreasures, :death, :specificVisibleTreasures, :specificHiddenTreasures
  private_class_method:new
  def initialize(aText, someLevels, someVisibleTreasures, someHiddenTreasures, 
someSpecificVisibleTreasures, someSpecificHiddenTreasures, death) 
    @text = aText
    @levels = someLevels
    @nVisibleTreasures = someVisibleTreasures
    @nHiddenTreasures = someHiddenTreasures
    @specificVisibleTreasures = someSpecificVisibleTreasures
    @specificHiddenTreasures = someSpecificHiddenTreasures
    @death = death
  end
  
  private_class_method :new
  
  def BadConsequence.newLevelNumberOfTreasures(aText, someLevels, 
someVisibleTreasures, someHiddenTreasures)
    new(aText, someLevels,someVisibleTreasures, someHiddenTreasures,[],Array.new,false)
  end
  
  def BadConsequence.newLevelSpecificTreasures (aText, someLevels,                          
someSpecificVisibleTreasures, someSpecificHiddenTreasures)
    new(aText, someLevels,0, 0,someSpecificVisibleTreasures,someSpecificHiddenTreasures,false)
  end
  
    def BadConsequence.newDeath (aText)
    new(aText, 0,0, 0,[],[],true)
  end

  
  def getText(text)
    @text
  end
  
  def getLevels(levels)
    @levels
  end
  
  def getNVisibleTreasures
    @nVisibleTreasures
  end
  
  def getNHiddenTreasures
    @nHiddenTreasures
  end
  
  def getDeath
    @death
  end
  
  def to_s
    "Text:#{@text} \n Niveles: #{@levels}\n Visible treasures #{@nVisibleTreasures}
\n Hidden treasures: #{@nHiddenTreasures}\n Death: #{@death}"
  end
  
end

File: PDOO/NapakalakiJava/BadConsequence.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package napakalaki;
import java.util.ArrayList;

/**
 *
 * @author adrian97
 */
public class BadConsequence {
    private String text;
    private int levels, nVisibleTreasures, nHiddenTreasures;
    private boolean death;
    
    public BadConsequence (String text, int levels, int nVisible, int nHidden){
        this.text = text;
        this.levels = levels;
        this.nVisibleTreasures = nVisibleTreasures;
        this.nHiddenTreasures = nHiddenTreasures; 
    }
    public BadConsequence(String text, boolean death){
        this.text = text;
        this.death = death;
    }
    private ArrayList<TreasureKind> specificVisibleTreasures = new ArrayList();
    private ArrayList<TreasureKind> specificHiddenTreasures = new ArrayList();

    public BadConsequence(String text, int levels, ArrayList<TreasureKind> tVisible,ArrayList<TreasureKind> tHidden){
        this.specificVisibleTreasures = tVisible;
        this.specificHiddenTreasures = tHidden;
    }
    

    public String getText(){
        return text;
    }
    public int getLevels(){
        return levels;
    }
    public int getNVisibleTreasures(){
        return nVisibleTreasures;
    }    
    public int getNHiddenTreasures(){
        return nHiddenTreasures;
    }
    public ArrayList getsHiddenTreasures(){
        
        return this.specificHiddenTreasures;
    }
    
    public ArrayList getsVisibleTreasures(){
        
        return this.specificVisibleTreasures;
    }
    public boolean getDeath(){
        return death;
    }
    
    public String toString(){
        return "Text = " + text +  " levels = " + Integer.toString(levels) + " Visible treasures = "
                + Integer.toString(nVisibleTreasures) + " Hidden treasures = " + Integer.toString(nHiddenTreasures);
    }
}
